dragnet/compat.py

```diff
@@ -8,20 +8,259 @@
     bytes_ = str
     unicode_ = unicode
     string_ = (str, unicode)
+    from itertools import izip as zip_
     import cPickle as pickle
     from StringIO import StringIO as bytes_io
 else:
     range_ = range
     bytes_ = bytes
     unicode_ = str
     string_ = (bytes, str)
+    zip_ = zip
     import pickle
     from io import BytesIO as bytes_io
 
+
+def str_cast(maybe_bytes, encoding='utf-8'):
+    """
+    Converts any bytes-like input to a string-like output, with respect to python version
+
+    Parameters
+    ----------
+    maybe_bytes : if this is a bytes-like object, it will be converted to a utf-8 string
```

Comment:
since `encoding` is a parameter, this doesn't necessarily have to be converted to a utf-8 string, right? if so, the wording should probably reflect it.